### PR TITLE
Add spider for Natal/RN

### PR DESCRIPTION
**Pull request by dannnylo**
In this PR we add the spider to collect data from [Natal/RN gazette](http://www.natal.rn.gov.br/dom/)

### CODE REVIEW

File: processing/data_collection/gazette/spiders/rn_natal.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        @url http://www.natal.rn.gov.br/dom/
+        @returns items 1
+        @scrapes date file_urls is_extra_edition municipality_id power scraped_at
+        """
+
+        for element in response.css(self.GAZETTE_ELEMENT_CSS):
+            url = response.urljoin(element.css('::attr(href)').extract_first())
+            link_text = element.css('::text').extract_first()
+            date = dateparser.parse(link_text.split(' - ')[-1], languages=['pt']).date()
+            extra_edition = ("Extra" in link_text)
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rn_natal.py]: Why the parenthesis? An extra comma (by accident) here may create an interesting bug hard to discover :-)

File: processing/data_collection/gazette/spiders/rn_natal.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import re
+import dateparser
+import w3lib.url
+
+from datetime import datetime
+from scrapy import Request, Spider, FormRequest
+from gazette.items import Gazette
+
+
+class RnNatalSpider(Spider):
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rn_natal.py]: Inherit from `BaseGazetteSpider`.

File: processing/data_collection/gazette/spiders/rn_natal.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    allowed_domains = ['www.natal.rn.gov.br']
+    name = 'rn_natal'
+
+    def start_requests(self):
+        base_url = 'http://www.natal.rn.gov.br/dom/'
+
+        for year in range(2015, datetime.now().year + 1):
+            for month in range(1, 13):
+                print([year, month])
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rn_natal.py]: If you want to log something, use `self.logger`. But I don't see the need to log this information here (except for development).

File: processing/data_collection/gazette/spiders/rn_natal.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    GAZETTE_ELEMENT_CSS = '#texto a'
+    MUNICIPALITY_ID = '2408102'
+
+    allowed_domains = ['www.natal.rn.gov.br']
+    name = 'rn_natal'
+
+    def start_requests(self):
+        base_url = 'http://www.natal.rn.gov.br/dom/'
+
+        for year in range(2015, datetime.now().year + 1):
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rn_natal.py]: If it is possible to get everything from 2003, there is no need to limit it here. Date filtering should be done in item pipeline already included in the project.

File: processing/data_collection/gazette/spiders/rn_natal.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            for month in range(1, 13):
+                print([year, month])
+                data = dict(ano=str(year), mes=str(month), list='Listar')
+                yield FormRequest(url=base_url, formdata=data, callback=self.parse)
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rn_natal.py]: `self.parse` is always the default callback for a `Request` or `FormRequest`. There is no need to declare it.

**Comment by dannnylo** [comment]
@rennerocha 
Fixed all comments. 
Thanks.

**Comment by jvanz** [comment]
ping

**Comment by victor-torres** [comment]
Do you need help with this pull request? Is it working?

**Comment by dannnylo** [comment]
@victor-torres If you can continue it will be a great help, two years ago I did this PR.

**Comment by victor-torres** [comment]
Thank you for your permission and previous work, @danilolutz. 
There was only a single issue with the spider since the website was rejecting month numbers with single digits. 
6 should be zero-filled as 06, for example.

@jvanz, commits on this pull request were moved to #192. This one could be closed.

**Comment by jvanz** [comment]
:+1:

**Pull request closed without merging**